src/wpantund/NCPInstanceBase-NetInterface.cpp

```diff
@@ -34,6 +34,37 @@
 using namespace nl;
 using namespace wpantund;
 
+void
+NCPInstanceBase::link_state_changed(bool isUp, bool isRunning)
+{
+	syslog(LOG_INFO, "Primary link state changed: UP=%d RUNNING=%d", isUp, isRunning);
+
+	// The big take away from this callback is "isUp",
+	// because theoretically we are the one who is in charge
+	// of "isRunning". We interpret isUp as meaning if
+	// we should autoconnect or not.
+
+	if (isUp) {
+		set_property(kWPANTUNDProperty_DaemonAutoAssociateAfterReset, true);
+
+		if (!mEnabled) {
+			set_property(kWPANTUNDProperty_DaemonEnabled, true);
+		} else if (get_ncp_state() == COMMISSIONED) {
+			set_property(kWPANTUNDProperty_InterfaceUp, true);
+		}
+	} else {
+		set_property(kWPANTUNDProperty_DaemonAutoAssociateAfterReset, false);
```

Comment:
It is ideally just supposed to be another way that this variable can be set. We might just retire this property entirely.